pkg/sypgp/sypgp.go

```diff
@@ -628,79 +610,57 @@ func FetchPubkey(fingerprint, keyserverURI, authToken string, noPrompt bool) (op
 	}
-
 	return el, nil
 }
 
-// doPushRequest prepares an HKP pks/add request
-func doPushRequest(w *bytes.Buffer, keyserverURI, authToken string) (*http.Request, error) {
-	v := url.Values{}
-	v.Set("keytext", w.String())
+func serializeEntity(e *openpgp.Entity, blockType string) (string, error) {
+	w := bytes.NewBuffer(nil)
 
-	u, err := url.Parse(keyserverURI)
+	wr, err := armor.Encode(w, blockType, nil)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	u.Path = "pks/add"
 
-	r, err := http.NewRequest(http.MethodPost, u.String(), strings.NewReader(v.Encode()))
-	if err != nil {
-		return nil, err
-	}
-	if authToken != "" {
-		r.Header.Set("Authorization", fmt.Sprintf("BEARER %s", authToken))
+	if err = e.Serialize(wr); err != nil {
+		wr.Close()
+		return "", err
 	}
-	r.Header.Set("User-Agent", useragent.Value())
-	r.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	wr.Close()
 
-	return r, nil
+	return w.String(), nil
 }
 
-// PushPubkey pushes a public key to a key server
-func PushPubkey(entity *openpgp.Entity, keyserverURI, authToken string) error {
-	w := bytes.NewBuffer(nil)
-	wr, err := armor.Encode(w, openpgp.PublicKeyType, nil)
-	if err != nil {
-		return err
-	}
+// PushPubkey pushes a public key to the Key Service.
+func PushPubkey(e *openpgp.Entity, keyserverURI, authToken string) error {
+	keyText, err := serializeEntity(e, openpgp.PublicKeyType)
 
-	err = entity.Serialize(wr)
+	// Get a Key Service client.
+	c, err := client.NewClient(&client.Config{
+		BaseURL:   keyserverURI,
+		AuthToken: authToken,
+	})
 	if err != nil {
 		return err
 	}
-	wr.Close()
 
-	r, err := doPushRequest(w, keyserverURI, authToken)
-	if err != nil {
-		return fmt.Errorf("error while preparing http request: %s", err)
-	}
+	// Push key to Key Service.
+	if err := c.PKSAdd(context.TODO(), keyText); err != nil {
+		if err, ok := err.(*jsonresp.Error); ok && err.Code == http.StatusUnauthorized {
 
-	resp, err := http.DefaultClient.Do(r)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
+			// The request failed with HTTP code unauthorized. Guide user to fix that.
+			authToken, err := helpAuthentication()
+			if err != nil {
+				return fmt.Errorf("could not obtain or install authentication token: %s", err)
+			}
 
-	// check if error is authentication failure and help user when it's the case
-	if resp.StatusCode == http.StatusUnauthorized {
-		token, err := helpAuthentication()
-		if err != nil {
-			return fmt.Errorf("could not obtain or install authentication token: %s", err)
-		}
-		// try request again
-		r, err := doPushRequest(w, keyserverURI, token)
-		if err != nil {
-			return fmt.Errorf("error while preparing http request: %s", err)
+			// Try to push key to Key Service again with new auth token.
+			c.AuthToken = authToken
+			if err := c.PKSAdd(context.TODO(), keyText); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("Key server did not accept PGP key: %v", err)
```

Comment:
`fmt.Errof(...)` should not begin with capital letter. Instead: `fmt.Errorf("key server [...]")`

Reply:
Will do.